Deduplication Using Lowercase for Keyword Generations

positiveVibes4all opened the pull request:
## Summary 
Keyword generations need to be unique, regardless of the casing that is used for the words.

decrapifier commented:
<!--
  2 failure:  The version was n..., The changelog has...
  0 warning: 
  
  1 markdown notices
  DangerID: danger-id-Danger;
-->

<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>The version was not updated. Please update the version.</td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>The changelog has not been updated. Please update the changelog.</td>
    </tr>
  </tbody>
</table>



## Coverage Report

> :+1: Test coverage is looking good.

|Files|% Stmts|% Branch|% Funcs|% Lines|Uncovered Lines||
|---|:-:|:-:|:-:|:-:|:-:|:-:|
|[src/dataStrategies.js](../blob/497e1b1a8b7f28724ff23fb53379d1e946055fce/src/dataStrategies.js)|88.68|100|83.33|88.68|[27](../blob/497e1b1a8b7f28724ff23fb53379d1e946055fce/src/dataStrategies.js#L27), [28](../blob/497e1b1a8b7f28724ff23fb53379d1e946055fce/src/dataStrategies.js#L28), [29](../blob/497e1b1a8b7f28724ff23fb53379d1e946055fce/src/dataStrategies.js#L29), [30](../blob/497e1b1a8b7f28724ff23fb53379d1e946055fce/src/dataStrategies.js#L30), [31](../blob/497e1b1a8b7f28724ff23fb53379d1e946055fce/src/dataStrategies.js#L31), [32](../blob/497e1b1a8b7f28724ff23fb53379d1e946055fce/src/dataStrategies.js#L32)|:white_check_mark:|
|[src/exampleTypes.js](../blob/497e1b1a8b7f28724ff23fb53379d1e946055fce/src/exampleTypes.js)|93.48|80|100|93.48|[10](../blob/497e1b1a8b7f28724ff23fb53379d1e946055fce/src/exampleTypes.js#L10), [11](../blob/497e1b1a8b7f28724ff23fb53379d1e946055fce/src/exampleTypes.js#L11), [12](../blob/497e1b1a8b7f28724ff23fb53379d1e946055fce/src/exampleTypes.js#L12)|:white_check_mark:|
|[src/index.js](../blob/497e1b1a8b7f28724ff23fb53379d1e946055fce/src/index.js)|94.74|93.33|100|94.74|[48](../blob/497e1b1a8b7f28724ff23fb53379d1e946055fce/src/index.js#L48), [49](../blob/497e1b1a8b7f28724ff23fb53379d1e946055fce/src/index.js#L49), [58](../blob/497e1b1a8b7f28724ff23fb53379d1e946055fce/src/index.js#L58), [59](../blob/497e1b1a8b7f28724ff23fb53379d1e946055fce/src/index.js#L59)|:white_check_mark:|

<details>
<summary>
and 7 more...
</summary>

|Files|% Stmts|% Branch|% Funcs|% Lines|Uncovered Lines||
|---|:-:|:-:|:-:|:-:|:-:|:-:|
|[src/utils.js](../blob/497e1b1a8b7f28724ff23fb53379d1e946055fce/src/utils.js)|84.95|94.59|100|84.95|[44](../blob/497e1b1a8b7f28724ff23fb53379d1e946055fce/src/utils.js#L44), [45](../blob/497e1b1a8b7f28724ff23fb53379d1e946055fce/src/utils.js#L45), [46](../blob/497e1b1a8b7f28724ff23fb53379d1e946055fce/src/utils.js#L46), [47](../blob/497e1b1a8b7f28724ff23fb53379d1e946055fce/src/utils.js#L47), [48](../blob/497e1b1a8b7f28724ff23fb53379d1e946055fce/src/utils.js#L48), [49](../blob/497e1b1a8b7f28724ff23fb53379d1e946055fce/src/utils.js#L49), [64](../blob/497e1b1a8b7f28724ff23fb53379d1e946055fce/src/utils.js#L64), [65](../blob/497e1b1a8b7f28724ff23fb53379d1e946055fce/src/utils.js#L65), [66](../blob/497e1b1a8b7f28724ff23fb53379d1e946055fce/src/utils.js#L66), [67](../blob/497e1b1a8b7f28724ff23fb53379d1e946055fce/src/utils.js#L67)...|:white_check_mark:|
|[src/__data__/imdb.js](../blob/497e1b1a8b7f28724ff23fb53379d1e946055fce/src/__data__/imdb.js)|100|100|100|100||:white_check_mark:|
|[src/provider-debug/index.js](../blob/497e1b1a8b7f28724ff23fb53379d1e946055fce/src/provider-debug/index.js)|100|100|100|100||:white_check_mark:|
|[src/provider-memory/date.js](../blob/497e1b1a8b7f28724ff23fb53379d1e946055fce/src/provider-memory/date.js)|94.55|94.44|60|94.55|[9](../blob/497e1b1a8b7f28724ff23fb53379d1e946055fce/src/provider-memory/date.js#L9), [10](../blob/497e1b1a8b7f28724ff23fb53379d1e946055fce/src/provider-memory/date.js#L10), [11](../blob/497e1b1a8b7f28724ff23fb53379d1e946055fce/src/provider-memory/date.js#L11), [64](../blob/497e1b1a8b7f28724ff23fb53379d1e946055fce/src/provider-memory/date.js#L64), [65](../blob/497e1b1a8b7f28724ff23fb53379d1e946055fce/src/provider-memory/date.js#L65), [66](../blob/497e1b1a8b7f28724ff23fb53379d1e946055fce/src/provider-memory/date.js#L66)|:white_check_mark:|
|[src/provider-memory/<br>exampleTypes.js](../blob/497e1b1a8b7f28724ff23fb53379d1e946055fce/src/provider-memory/exampleTypes.js)|91.26|92|82.35|91.26|[16](../blob/497e1b1a8b7f28724ff23fb53379d1e946055fce/src/provider-memory/exampleTypes.js#L16), [17](../blob/497e1b1a8b7f28724ff23fb53379d1e946055fce/src/provider-memory/exampleTypes.js#L17), [18](../blob/497e1b1a8b7f28724ff23fb53379d1e946055fce/src/provider-memory/exampleTypes.js#L18), [96](../blob/497e1b1a8b7f28724ff23fb53379d1e946055fce/src/provider-memory/exampleTypes.js#L96), [97](../blob/497e1b1a8b7f28724ff23fb53379d1e946055fce/src/provider-memory/exampleTypes.js#L97), [98](../blob/497e1b1a8b7f28724ff23fb53379d1e946055fce/src/provider-memory/exampleTypes.js#L98), [99](../blob/497e1b1a8b7f28724ff23fb53379d1e946055fce/src/provider-memory/exampleTypes.js#L99), [100](../blob/497e1b1a8b7f28724ff23fb53379d1e946055fce/src/provider-memory/exampleTypes.js#L100), [101](../blob/497e1b1a8b7f28724ff23fb53379d1e946055fce/src/provider-memory/exampleTypes.js#L101)|:white_check_mark:|
|[src/provider-memory/<br>index.js](../blob/497e1b1a8b7f28724ff23fb53379d1e946055fce/src/provider-memory/index.js)|100|75|100|100||:white_check_mark:|
|[src/provider-memory/<br>results.js](../blob/497e1b1a8b7f28724ff23fb53379d1e946055fce/src/provider-memory/results.js)|94.44|66.67|100|94.44|[14](../blob/497e1b1a8b7f28724ff23fb53379d1e946055fce/src/provider-memory/results.js#L14)|:white_check_mark:|
</details>
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 497e1b1a8b7f28724ff23fb53379d1e946055fce
</p>

Commit:
Deduplicate with toLower 

- Dedup with toLower function so that all duplicates get remove regardless of case

Commit:
BUMP

lint-action committed:
Fix code style issues with Prettier

Commit:
Remove Duplicated Logic

daedalus28 commented on packages/react/src/exampleTypes/ExpandableTagsQuery/index.js:
```diff
@@ -132,8 +133,8 @@ let ExpandableTagsQuery = ({
               _.reject(
-                _.includes(_, _.memoize(_.keys)(node.context.tags)),
-                _.keys(node.context.keywordGenerations)
+                _.includes(_, _.memoize(keysToLower)(node.context.tags)),
```
It occurs to me that the memoize makes no difference here since the function is called right away. Removing it will simplify the code and might actually slightly improve performance

Commit:
Merge remote-tracking branch 'origin/feature/Keyword-Generations-Dedup-With-LowerCase' into feature/Keyword-Generations-Dedup-With-LowerCase

Commit:
Refactor

positiveVibes4all committed:
Merge branch 'main' into feature/Keyword-Generations-Dedup-With-LowerCase
